panel/lxqtpanel.cpp

```diff
@@ -374,7 +372,7 @@ int LXQtPanel::getReserveDimension()
     return mHidable ? PANEL_HIDE_SIZE : qMax(PANEL_MINIMUM_SIZE, mPanelSize);
 }
 
-void LXQtPanel::setPanelGeometry(bool animate)
+void LXQtPanel::setPanelGeometry(bool animate, bool firstTime)
```

Comment:
> Without it, there would be a mess. Since one test is worth a million words, try it ;

I tested it back then and also this time. I see the same available desktop geometry with or without the `firstTime` flag.

Consider this patch:
```diff
diff --git a/panel/lxqtpanel.cpp b/panel/lxqtpanel.cpp
index 90bae87..49885e1 100644
--- a/panel/lxqtpanel.cpp
+++ b/panel/lxqtpanel.cpp
@@ -201,14 +201,14 @@ LXQtPanel::LXQtPanel(const QString &configGroup, LXQt::Settings *settings, QWidg
 
     loadPlugins();
 
-    show();
-
     // show it the first time, despite setting
     if (mHidable)
     {
-      showPanel(true);
+      showPanel(false);
       QTimer::singleShot(PANEL_HIDE_FIRST_TIME, this, SLOT(hidePanel()));
     }
+
+    show();
 }
 
 /************************************************
@@ -377,7 +377,7 @@ int LXQtPanel::getReserveDimension()
     return mHidable ? PANEL_HIDE_SIZE : qMax(PANEL_MINIMUM_SIZE, mPanelSize);
 }
 
-void LXQtPanel::setPanelGeometry(bool animate, bool firstTime)
+void LXQtPanel::setPanelGeometry(bool animate)
 {
     const QRect currentScreen = QApplication::desktop()->screenGeometry(mActualScreenNum);
     QRect rect;
@@ -417,14 +417,14 @@ void LXQtPanel::setPanelGeometry(bool animate, bool firstTime)
         // Vert .......................
         if (mPosition == ILXQtPanel::PositionTop)
         {
-            if (mHidden && !firstTime)
+            if (mHidden)
                 rect.moveBottom(currentScreen.top() + PANEL_HIDE_SIZE);
             else
                 rect.moveTop(currentScreen.top());
         }
         else
         {
-            if (mHidden && !firstTime)
+            if (mHidden)
                 rect.moveTop(currentScreen.bottom() - PANEL_HIDE_SIZE);
             else
                 rect.moveBottom(currentScreen.bottom());
@@ -465,14 +465,14 @@ void LXQtPanel::setPanelGeometry(bool animate, bool firstTime)
         // Horiz ......................
         if (mPosition == ILXQtPanel::PositionLeft)
         {
-            if (mHidden && !firstTime)
+            if (mHidden)
                 rect.moveRight(currentScreen.left() + PANEL_HIDE_SIZE);
             else
                 rect.moveLeft(currentScreen.left());
         }
         else
         {
-            if (mHidden && !firstTime)
+            if (mHidden)
                 rect.moveLeft(currentScreen.right() - PANEL_HIDE_SIZE);
             else
                 rect.moveRight(currentScreen.right());
@@ -943,7 +943,7 @@ bool LXQtPanel::event(QEvent *event)
         event->ignore();
         //no break intentionally
     case QEvent::Enter:
-        showPanel();
+        showPanel(mAnimationTime > 0);
         break;
 
     case QEvent::Leave:
@@ -1164,7 +1164,7 @@ void LXQtPanel::userRequestForDeletion()
     emit deletedByUser(this);
 }
 
-void LXQtPanel::showPanel(bool firstTime)
+void LXQtPanel::showPanel(bool animate)
 {
     if (mHidable)
     {
@@ -1172,7 +1172,7 @@ void LXQtPanel::showPanel(bool firstTime)
         if (mHidden)
         {
             mHidden = false;
-            setPanelGeometry(mAnimationTime > 0, firstTime);
+            setPanelGeometry(mAnimationTime > 0 && animate);
         }
     }
 }
@@ -1192,7 +1192,7 @@ void LXQtPanel::hidePanelWork()
         if (!mStandaloneWindows->isAnyWindowShown())
         {
             mHidden = true;
-            setPanelGeometry(mAnimationTime > 0, false);
+            setPanelGeometry(mAnimationTime > 0);
         } else
         {
             mHideTimer.start();
diff --git a/panel/lxqtpanel.h b/panel/lxqtpanel.h
index 3a6897b..d893476 100644
--- a/panel/lxqtpanel.h
+++ b/panel/lxqtpanel.h
@@ -243,11 +243,11 @@ public slots:
      * Stops the QTimer mHideTimer. This it NOT the same as QWidget::show()
      * because hiding the panel in LXQt is done by making it very thin. So
      * this method in fact restores the original size of the panel.
-     * \param firstTime flag if the panel is shown for the first time.
+     * \param animate flag for the panel show-up animation disabling (\sa mAnimationTime).
      *
      * \sa mHidable, mHidden, mHideTimer, hidePanel(), hidePanelWork()
      */
-    void showPanel(bool firstTime = false);
+    void showPanel(bool animate);
     /**
      * @brief Hides the panel (delayed) by starting the QTimer mHideTimer.
      * When this timer times out, hidePanelWork() will be called. So this
@@ -496,9 +496,8 @@ private:
      * on the screen) of the panel. Considers alignment, position, if the panel
      * is hidden and if its geometry should be set with animation.
      * \param animate flag if showing/hiding the panel should be animated.
-     * \param firstTime flag if the panel geometry is set for the first time.
      */
-    void setPanelGeometry(bool animate = false, bool firstTime = false);
+    void setPanelGeometry(bool animate = false);
     /**
      * @brief Calculates the height of the panel if it is horizontal or the
      * width if the panel is vertical. Considers if the panel is hidden and
```

... with it I believe the first time show-up (and setting the available geometry) is correct.